DeliveryApi/cdk/src/main/java/com/ilmlf/delivery/api/ApiStack.java

```diff
@@ -0,0 +1,507 @@
+import software.amazon.awscdk.core.Environment;
+import software.amazon.awscdk.core.Stack;
+import software.amazon.awscdk.core.StackProps;
+import software.amazon.awscdk.customresources.Provider;
+import software.amazon.awscdk.customresources.ProviderProps;
+import software.amazon.awscdk.services.apigateway.ApiDefinition;
+import software.amazon.awscdk.services.apigateway.CfnAccount;
+import software.amazon.awscdk.services.apigateway.CfnAccountProps;
+import software.amazon.awscdk.services.apigateway.Cors;
+import software.amazon.awscdk.services.apigateway.SpecRestApi;
+import software.amazon.awscdk.services.ec2.ISecurityGroup;
+import software.amazon.awscdk.services.ec2.IVpc;
+import software.amazon.awscdk.services.ec2.SecurityGroup;
+import software.amazon.awscdk.services.ec2.Vpc;
+import software.amazon.awscdk.services.ec2.VpcLookupOptions;
+import software.amazon.awscdk.services.iam.CfnRole;
+import software.amazon.awscdk.services.iam.Effect;
+import software.amazon.awscdk.services.iam.ManagedPolicy;
+import software.amazon.awscdk.services.iam.PolicyStatement;
+import software.amazon.awscdk.services.iam.PolicyStatementProps;
+import software.amazon.awscdk.services.iam.Role;
+import software.amazon.awscdk.services.iam.RoleProps;
+import software.amazon.awscdk.services.iam.ServicePrincipal;
+import software.amazon.awscdk.services.lambda.CfnFunction;
+import software.amazon.awscdk.services.lambda.Code;
+import software.amazon.awscdk.services.lambda.Function;
+import software.amazon.awscdk.services.lambda.FunctionProps;
+import software.amazon.awscdk.services.lambda.Runtime;
+import software.amazon.awscdk.services.logs.LogGroup;
+import software.amazon.awscdk.services.logs.LogGroupProps;
+import software.amazon.awscdk.services.logs.RetentionDays;
+import software.amazon.awscdk.services.s3.assets.AssetOptions;
+import software.amazon.awscdk.services.sam.CfnApi;
+import software.amazon.awscdk.services.sam.CfnApiProps;
+
+public class ApiStack extends Stack {
+
+  private final IVpc dbVpc;
+  private final ISecurityGroup dbSg;
+  private final Role lambdaRdsProxyRoleWithIam;
+
+  @lombok.Builder
+  @Data
+  public static class ApiStackProps implements StackProps {
+    private String dbProxyEndpoint;
+    private String dbProxyArn;
+    private String dbEndpoint;
+    private String dbAdminSecretName;
+    private String dbAdminSecretArn;
+    private String dbUserSecretName;
+    private String dbUserSecretArn;
+    private String dbUser;
+
+    /**
+     * VPC that the database is deployed to.
+     */
+    private Vpc dbVpc;
+    private SecurityGroup dbSg;
+
+    private String dbRegion;
+    private Environment env;
+  }
+
+  public ApiStack(final Construct scope, final String id, final ApiStackProps props)
+      throws IOException {
+    super(scope, id, props);
+
+    // Role for Lambda to connect to RDS Proxy via IAM authentication
+    lambdaRdsProxyRoleWithIam = createLambdaRdsProxyRoleWithIam(props.getDbUser());
```

Comment:
shouldn't it be this.lambdaRdsProxyRoleWithIam to make it clear it is a class attribute?

Reply:
It is not clear to me if it will be used to connect to the db directly or through the RDS Proxy too.

Reply:
From my understanding of the code, it will be use to populate data in the db through a direct connection to the database so the variable name is misleading.

Reply:
done